crates/ironrdp-svc/src/lib.rs

```diff
@@ -36,22 +37,17 @@ pub enum CompressionCondition {
 /// There are at most 31 (optional) static virtual channels that can be created for a single connection, for a
 /// total of 32 static channels when accounting for the non-optional I/O channel.
 pub trait StaticVirtualChannel: AsAny + fmt::Debug + Send + Sync {
-    /// Returns the name of the `StaticVirtualChannel` that is created by the `make_static_channel` method
+    /// Returns the name of the `StaticVirtualChannel`
     fn channel_name(&self) -> ChannelName;
 
     /// Defines which compression flag should be sent along the Channel Definition Structure (CHANNEL_DEF)
     fn compression_condition(&self) -> CompressionCondition {
         CompressionCondition::Never
     }
 
-    /// Processes a complete block (chunks must be assembled by calling code)
-    fn process(
-        &mut self,
-        initiator_id: StaticChannelId,
-        channel_id: StaticChannelId,
-        payload: &[u8],
-        output: &mut WriteBuf,
-    ) -> PduResult<()>;
+    /// Processes a payload received on the virtual channel.
+    /// Returns a list of PDUs to be sent back to the client.
+    fn process(&mut self, payload: &[u8]) -> PduResult<Vec<Box<dyn PduEncode>>>;
```

Comment:
I was initially a bit skeptical about returning a `Vec` of trait objects, but it appears to have a few nice properties:
- the trait is more straightforward to implement downstream, and
- it actually opens up the door to further optimizations down the road: we can actually perform inline encoding of the packet into the final output buffer given that `PduEncode::size` returns a value smaller that the maximum chunk size for instance.
Let’s roll with that and see where we end up.

Reply:
Yep, I agree too!